docs/src/pages/ThemingSpecification.mdx

```diff
@@ -0,0 +1,301 @@
+| Adding new design term in the parent requires re-composing all slots components                                   | NO                                          | We can just override styles in the slot's styles inside the parents                                                                            | YES                                      | We need to re-compose if we want new design terms in the slots components                                               |
+| Requires knowledge of the internals of another component, which creates coupling                                  | PARTIALLY                                   | The parent can define styles only for the root (no other slots), but it needs to know some details for the root's styles of the slot component | NO                                       | All styles are defined in the composed component                                                                        |
+| Easy to swap a slot into another component which might want all its own styles.                                   | NO                                          |                                                                                                                                                | YES                                      |
+
+##### Variants
+
+Variants are just mechanism for allowing users do add or remove some design terms from their components.
+
+In order to enable this, the compose mechanism should provide to the users adding/removing some of the design terms. This can be easily done, by allowing the user to define mapping from props to design terms when they define the composed components.
+
+```
+export interface FluentButton extends ButtonProps {
+  primary?: boolean
+  circular?: boolean
+}
+
+// these code snippet is just illustrating a possible
+// mechanism of how the design tokens can be created
+// based on the props
+const FluentButton = compose(Button, {
+  displayName: 'FluentButton',
+  mapPropsToTerms: props => ({
+    primary: props.primary,
+    circular: props.circular,
+    hasIcon: !!props.icon,
+  }),
+})
+```
+
+##### Animations
+
+Animation should be defined as part of the theme, but they should not be all loaded if they are not used. Child themes should be able to customize them.
+
+```
+// TBD
+```
+
+##### Icons
+
+Icons should be defined as part of the theme, but they should not loaded if not used. Child themes should be able to customize them.
+
+```
+// TBD
+```
+
+##### Files and packaging
+
+```
+// themes/fluent/globalTokens.ts
+
+const globalTokens = {
+  colors: {
+    brandForeground: 'white',
+    brandBackground: 'red',
+
+    defaultBackground: 'white',
+    defaultForeground: 'black',
+  }
+}
+
+export default globalTokens
```

Comment:
If we will use Styleguide it will be a list of separate constants:

```tsx
export colorBrandForeground = 'white'
export colorBrandBackground = 'red'
```